ambari-server/src/main/java/org/apache/ambari/server/topology/MpackInstance.java

```diff
@@ -117,37 +121,59 @@ public void setUrl(String url) {
     this.url = url;
   }
 
-  public BlueprintMpackInstanceEntity toEntity() {
-    BlueprintMpackInstanceEntity mpackEntity = new BlueprintMpackInstanceEntity();
-    mpackEntity.setMpackUri(url);
-    mpackEntity.setMpackName(mpackName);
-    mpackEntity.setMpackVersion(mpackVersion);
-    Collection<BlueprintMpackConfigEntity> mpackConfigEntities =
-      BlueprintImpl.toConfigEntities(configuration, BlueprintMpackConfigEntity::new);
-    mpackConfigEntities.forEach( configEntity -> configEntity.setMpackInstance(mpackEntity) );
-    mpackEntity.setConfigurations(mpackConfigEntities);
+  /**
+   * Converts the mpack instance to a {@link BlueprintMpackInstanceEntity}
+   * @param blueprintEntity the blueprint entity will be associated to
+   * @return the resulting entity
+   */
+  public BlueprintMpackInstanceEntity toMpackInstanceEntity(BlueprintEntity blueprintEntity) {
+    BlueprintMpackInstanceEntity mpackInstanceEntity = new BlueprintMpackInstanceEntity();
+    mpackInstanceEntity.setBlueprint(blueprintEntity);
+    setCommonProperties(mpackInstanceEntity);
+    return mpackInstanceEntity;
+  }
+
+  /**
+   * Converts the mpack instance to a {@link TopologyRequestMpackInstanceEntity}
+   * @param topologyRequestEntity the topology request entity will be associated to
+   * @return the resulting entity
+   */
+  public TopologyRequestMpackInstanceEntity toMpackInstanceEntity(TopologyRequestEntity topologyRequestEntity) {
+    TopologyRequestMpackInstanceEntity mpackInstanceEntity = new TopologyRequestMpackInstanceEntity();
+    mpackInstanceEntity.setTopologyRequest(topologyRequestEntity);
+    setCommonProperties(mpackInstanceEntity);
+    return mpackInstanceEntity;
+  }
+
+  private void setCommonProperties(MpackInstanceEntity mpackInstanceEntity) {
```

Comment:
Documentation of which properties are being set for the bi-directional relationship.